topics/data-science/tutorials/snakemake/tutorial.md

```diff
@@ -0,0 +1,1714 @@
+{: .code-out}
+
+Now that you have seen a few rules, let's write the rest.
+
+> ### {% icon hands_on %} Hands-on: Save this Snakefile to the filesystem
+> From here on you are going to finish writing this Snakefile on your own! We'll give you the bits you have seen up until now, and future tasks will require you to add your own rules and get to actually test out the pipeline!
+>
+> ```Snakemake
+> rule download:
+> 	output:
+> 		"{sample}.fq.gz"
+> 	shell:
+> 		"wget https://zenodo.org/record/5562251/files/{wildcards.sample}.fq.gz -O {output}"
+>
+> rule download_genome:
+> 	output:
+> 		"GCA_000017985.1_ASM1798v1_genomic.fna.gz"
+> 	shell:
+> 		"wget https://zenodo.org/record/5562251/files/GCA_000017985.1_ASM1798v1_genomic.fna.gz -O {output}"
+>
+> rule fastqc:
+> 	input:
+> 		"{sample}.fq.gz"
+> 	output:
+> 		"{sample}_fastqc.html"
+> 	shell:
+> 		"fastqc {input} --outdir fastqc/"
+>
+> rule trimmomatic:
+> 	input:
+> 		r1="{sample}_1.fq.gz",
+> 		r2="{sample}_2.fq.gz"
+> 	output:
+> 		o1="{sample}_1.trim.fq",
+> 		o2="{sample}_2.trim.fq",
+> 		o1un="{sample}_1un.trim.fq",
+> 		o2un="{sample}_2un.trim.fq"
+> 	shell:
+> 		"trimmomatic PE "
+> 		"{input.r1} {input.r2} "
+> 		"{output.o1} {output.o1un} "
+> 		"{output.o2} {output.o2un} "
+> 		"SLIDINGWINDOW:4:20 MINLEN:25 "
+> 		"ILLUMINACLIP:NexteraPE-PE.fa:2:40:15"
+> ```
+{: .hands_on}
+
+> ### {% icon hands_on %} Hands-on: Install Snakemake
+> We're about to start doing things really with snakemake, so, it's time to install it.
+> 1. [Install Miniconda](https://docs.conda.io/en/latest/miniconda.html)
+> 2. Once you've done that and activated conda (you should see `(base)` in your terminal prompt), run `conda install snakemake`
+{: .hands_on}
+
+
+> ### {% icon hands_on %} Hands-on: Try running snakemake!
+>
+> 1. Try running snakemake
+>
+>    > ### {% icon code-in %} CLI
+>    > ```
+>    > snakemake -c1
+>    > ```
+>    {: .code-in}
+>
+>    > ### {% icon code-out %} Output
+>    > ```
+>    > Building DAG of jobs...
+>    > WorkflowError:
+>    > Target rules may not contain wildcards. Please specify concrete files or a rule without wildcards at the command line, or have a rule without wildcards at the very top of your workflow (e.g. the typical "rule all" which just collects all results you want to generate in the end).
+>    > ```
+>    {: .code-out}
+>
+>    Wait, that didn't work! This is the `all` rule we also saw in the makefile. But we can use Snakemake to build individual rule outputs, so let's try that now.
+>
+> 2. Run snakemake to download `SRR2584866_1.fq.gz` and `SRR2584866_2.fq.gz`
+>
+>    > ### {% icon code-in %} CLI
+>    > ```
+>    > snakemake -c1 SRR2584866_1.fq.gz SRR2584866_2.fq.gz
```

Comment:
I already went through the tutorial to test all commands and it work. but i have one error which keeps bugging me and i dont know how to solve.


(snakemake_test) thor@asgard:~/snakemake_test$ snakemake -c1 SRR2584866_1.fq.gz SRR2584866_2.fq.gz --jobs 1


Building DAG of jobs...
Provided cluster nodes: 1
Job counts:
        count   jobs
        2       download
        2

rule download:
    output: SRR2584866_2.fq.gz
    jobid: 0
    wildcards: sample=SRR2584866_2

/bin/sh: 1: 1: not found
Error submitting jobscript (exit code 127):

Will exit after finishing currently running jobs.
Exiting because a job execution failed. Look above for error message

Reply:
the Snakefile is directly copied from the tutorial. 

i tried installing various snakemake conda enviorments.
One time this acutal worked tho.

@bazante and i are still figuring out why this occours

Reply:
That should definitely work, can you re-run with `-vvvv` or so to make it more verbose (I think that's the right syntax). 

Also could you please put code in codeblocks (three ` backticks on a single line, code, then another line with just three backticks) to make it a bit easier to read.

Reply:
Hi Helena,

I tried with -vvvv but it is not really the output i expected

```(snakemake) thor@asgard:~/snakemake_test$ snakemake -c1 SRR2584866_1.fq.gz SRR2584866_2.fq.gz --jobs 1 -vvvv
4.3.0
```

Reply:
Hi i double checked with the other linux system.
Where Snakemake worked, was it version 6.10.0.
So probably lay there the issue?

Reply:
Ahhh `-v` for version :joy: ok, clearly also you have a way outdated snakemake, bug found! We should add a tip box at the top "Tested with Snakemake 6.10.0" or so.